client/src/components/file-field/index.tsx

```diff
@@ -0,0 +1,74 @@
+'use client';
+import { Field, FieldProps } from 'formik';
+import { useTranslations } from 'next-intl';
+
+import { Input } from '../input';
+import { FileInputProps, InputOnChangeEventType } from '../input/types';
+
+export const FileField = ({
+  name,
+  label,
+  accept,
+  maxSize,
+  ...props
+}: FileInputProps) => {
+  const errors = useTranslations('errors.file');
+
+  return (
+    <Field name={name}>
+      {({ meta, form, field: { value, ...fieldProps } }: FieldProps) => {
+        const onChange = async (e: InputOnChangeEventType) => {
+          if (typeof e === 'string') {
+            await form.setFieldValue(name, '');
+
+            return;
+          }
+
+          const target = e?.target as HTMLInputElement;
+          const files = target.files;
+          const file = files?.[0];
+
+          if (!file) {
+            return;
+          }
+          const extension = file.name.split('.').pop()?.toLowerCase();
+
+          if (accept && !!extension) {
+            const isExtensionAccept = accept.includes(extension);
+
+            if (!isExtensionAccept) {
+              form.setFieldError(name, `${errors('extensions')} ${accept}`);
+              form.setFieldTouched(name, true, false);
+
+              return;
+            }
+          }
+
+          if (maxSize && file.size / (1024 * 1024) > maxSize) {
```

Comment:
I'd extract both those if conditions to a separate functions

Reply:
In this case, putting it into a separate function will create a large number of bolerplate code and additional handlers and typing.